upgrade/upgrade-provider.go

```diff
@@ -315,6 +321,17 @@ func UpgradeProvider(ctx Context, repoOrg, repoName string) error {
 		}()
 	}
 
+	steps = append(steps, step.Computed(func() step.Step {
+		// No upgrade was planned, so exit
+		if tfSDKTargetSHA == "" {
+			return nil
+		}
+		return step.Combined("Update Plugin SDK",
+			getLatestTFPluginSDKReplace(ctx, repo, &tfSDKTargetSHA),
+			step.Cmd(exec.CommandContext(ctx, "go", "mod", "tidy")).In(repo.providerDir()),
```

Comment:
Yes. `getLatestTFPluginSDKReplace` leaves `go.mod` in a valid but unresolved state. We need to run `go mod tidy` to resolve the SHA into a pseudo version so other `go` commands will work as expected.

We can't even parse `go.mod` when unresolved, since we don't include a resolver in our parsers.